agent/agent_endpoint_test.go

```diff
@@ -1342,6 +1344,126 @@ func TestAgent_Reload(t *testing.T) {
 	}
 }
 
+// TestAgent_ReloadDoesNotTriggerWatch Ensure watches not triggered after reload
+// see https://github.com/hashicorp/consul/issues/7446
+func TestAgent_ReloadDoesNotTriggerWatch(t *testing.T) {
+	t.Parallel()
+	dc1 := "dc1"
+	checkInterval := "1s" // This value ensure we could detect as transient critical state
+	tmpFileRaw, err := ioutil.TempFile("", "rexec")
+	if err != nil {
```

Comment:
opt: these can be `require.NoError(t, err)`

Reply:
DONE